core/src/main/java/com/kin/ecosystem/core/data/internal/ConfigurationImpl.java

```diff
@@ -81,16 +95,40 @@ public ApiClient getDefaultApiClient() {
 				defaultApiClient.addInterceptor(new Interceptor() {
 					@Override
 					public Response intercept(Chain chain) throws IOException {
-						Request originalRequest = chain.request();
-						if (isCanProceed(originalRequest)) {
+						Request originalRequest = blockchainSource == null ?
+							chain.request() :
+							chain.request() // new request with the BV version header
+								.newBuilder()
+								.addHeader(HEADER_BLOCKCHAIN_VERSION, blockchainSource.getBlockchainVersion().getVersion())
+								.build();
```

Comment:
if `getDefaultApiClient` will be called before  `blockchainSource ` is set, it will crash.